pkg/ebpf/tracer_test.go

```diff
@@ -1545,3 +1545,49 @@ func TestDNSStatsForInvalidDomain(t *testing.T) {
 func TestDNSStatsForTimeout(t *testing.T) {
 	testDNSStats(t, "golang.org", 0, 0, 1, "1.2.3.4")
 }
+
+func TestConntrackExpiration(t *testing.T) {
+	cmd := exec.Command("../network/netlink/testdata/setup_dnat.sh")
+	if out, err := cmd.CombinedOutput(); err != nil {
+		t.Errorf("setup command output: %s", string(out))
+	}
+	defer teardown(t)
+
+	tr, err := NewTracer(NewDefaultConfig())
+	require.NoError(t, err)
+	defer tr.Stop()
+
+	// Warm-up tracer state
+	_ = getConnections(t, tr)
+
+	// The random port is necessary to avoid flakiness in the test. Running the the test multiple
+	// times can fail if binding to the same port since Conntrack might not emit NEW events for the same tuple
+	rand.Seed(time.Now().UnixNano())
+	port := 5430 + rand.Intn(100)
+	server := NewUDPServerOnAddress(fmt.Sprintf("1.1.1.1:%d", port), func([]byte, int) []byte {
+		return nil
+	})
+	doneChan := make(chan struct{})
+	server.Run(doneChan, clientMessageSize)
+
+	c, err := net.Dial("udp", fmt.Sprintf("2.2.2.2:%d", port))
+	require.NoError(t, err)
+	defer c.Close()
+	_, err = c.Write([]byte("ping"))
+	require.NoError(t, err)
+
+	// Give enough time for conntrack cache to be populated
+	time.Sleep(100 * time.Millisecond)
```

Comment:
is there a way to do this without sleeping? it makes the tests flaky.

Reply:
It's an integration test with a lot of aync stuff going on, so I'm afraid there isn't much we can do 🤷‍♂️